bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/discovery/TouchWandUnitDiscoveryService.java

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.binding.touchwand.internal.discovery;
+
+import static org.openhab.binding.touchwand.internal.TouchWandBindingConstants.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.touchwand.internal.TouchWandBridgeHandler;
+import org.openhab.binding.touchwand.internal.TouchWandUnitStatusUpdateListener;
+import org.openhab.binding.touchwand.internal.dto.TouchWandUnitData;
+import org.openhab.binding.touchwand.internal.dto.TouchWandUnitFromJson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link TouchWandUnitDiscoveryService} Discovery service for TouchWand units.
+ *
+ * @author Roie Geron - Initial contribution
+ */
+@NonNullByDefault
+public class TouchWandUnitDiscoveryService extends AbstractDiscoveryService {
+
+    private static final int SEARCH_TIME_SEC = 10;
+    private static final int SCAN_INTERVAL_SEC = 60;
+    private static final int LINK_DISCOVERY_SERVICE_INITIAL_DELAY_SEC = 5;
+    private static final String[] CONNECTIVITY_OPTIONS = { "zwave", "knx" };
+    private final TouchWandBridgeHandler touchWandBridgeHandler;
+    private final Logger logger = LoggerFactory.getLogger(TouchWandUnitDiscoveryService.class);
+
+    private @Nullable ScheduledFuture<?> scanningJob;
+    private List<TouchWandUnitStatusUpdateListener> listeners = new ArrayList<>();
+
+    public TouchWandUnitDiscoveryService(TouchWandBridgeHandler touchWandBridgeHandler) {
+        super(SUPPORTED_THING_TYPES_UIDS, SEARCH_TIME_SEC, true);
+        this.touchWandBridgeHandler = touchWandBridgeHandler;
+        removeOlderResults(getTimestampOfLastScan(), touchWandBridgeHandler.getThing().getUID());
+        this.activate();
+    }
+
+    @Override
+    protected void startScan() {
+        if (touchWandBridgeHandler.touchWandClient == null) {
```

Comment:
This cannot be null, you can remove the null check to eliminate the compiler warning.